From d23bfe87b8692ab71cc0df8ceb0d3361b098732b Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 13 Mar 2024 10:28:11 +0100
Subject: [PATCH 1094/1141] dt-bindings: soc: st: add RISAF documentation

Add documentation on the RISAF peripheral that is a memory firewall.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I045e67dbd8b8e43d56b00c66503df91352f682cf
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/366282
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/soc/st/st,stm32mp25-risaf.yaml   | 65 +++++++++++++++++++
 1 file changed, 65 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/st/st,stm32mp25-risaf.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/st/st,stm32mp25-risaf.yaml
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/st/st,stm32mp25-risaf.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32 Resource isolation peripheral unit for address space protection
+       (full version)
+
+maintainers:
+  - Gatien Chevallier <gatien.chevallier@foss.st.com>
+
+description: |
+  The RIF (resource isolation framework) is a comprehensive set of hardware
+  blocks designed to enforce and manage isolation of STM32 hardware resources,
+  like memories and peripherals.
+  Through RISAF registers, a trusted domain, or the domain to whom the
+  configuration has been delegated, assigns memory regions and subregions to
+  one or more security domains (secure, privilege, compartment).
+  RISAF managing DRR includes the DDR memory cipher engine (DDRMCE) feature.
+
+properties:
+  compatible:
+    enum:
+      - st,stm32mp25-risaf
+      - st,stm32mp25-risaf-enc
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  st,mem-map:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    minItems: 2
+    maxItems: 4
+    description:
+      Defines the memory address range covered by the RISAF instance.
+
+  memory-region:
+    minItems: 1
+    maxItems: 16
+    description: |
+      Phandle to a node describing memory areas to be configured by the RISAF.
+      These regions cannot overlap. A zone must be within st,mem-map range.
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - st,mem-map
+
+additionalProperties: false
+
+examples:
+  - |
+        #include <dt-bindings/clock/stm32mp25-clks.h>
+
+        risaf4: risaf@420d0000 {
+          compatible = "st,stm32mp25-risaf-enc";
+          reg = <0x420d0000 0x1000>;
+          clocks = <&rcc CK_BUS_RISAF4>;
+          st,mem-map = <0x0 0x80000000 0x1 0x00000000>;
+        };
